src/signer/canister/src/sign/ecdsa_api.rs

```diff
@@ -0,0 +1,44 @@
+use ic_cdk::api::management_canister::ecdsa::{
+    ecdsa_public_key, sign_with_ecdsa, EcdsaCurve, EcdsaKeyId, EcdsaPublicKeyArgument,
+    SignWithEcdsaArgument,
+};
+
+pub async fn get_ecdsa_signature(
```

Comment:
Given that the canister uses only one key name, configured in the init arg, I estimate that it would probably simplify the codebase if we get the key name from the config here rather than taking it as an argument.  What do you think?

Reply:
I wasn't sure when to pass the key name or when to read from the state. I'm happy to red also here, yes.

It will simplify a few functions.